Core/ModuleInstaller.cs

```diff
@@ -889,29 +889,41 @@ private void Uninstall(string modName)
         /// <summary>
         /// Takes a collection of directories and adds all parent directories within the GameData structure.
         /// </summary>
-        /// <param name="directories">The collection of directory path strings to examine </param>
-        private HashSet<string> AddParentDirectories(HashSet<string> directories )
+        /// <param name="directories">The collection of directory path strings to examine</param>
+        public HashSet<string> AddParentDirectories(HashSet<string> directories)
         {
-            var newDirectories = new HashSet<string>();
-            foreach (string directory in directories)
+            if (directories == null || directories.Count == 0)
             {
-                if ( directory.Contains(ksp.GameData()))
+                return new HashSet<string>();
+            }
+
+            var gameData = KSPPathUtils.NormalizePath(ksp.GameData());
+            return directories
+                .Where(dir => !string.IsNullOrWhiteSpace(dir))
+                // normalize before we deduplicate this list
+                .Select(KSPPathUtils.NormalizePath)
+                // remove any duplicate paths
+                .Distinct()
+                // get rid of paths that are above GameData
+                .Where(dir => dir.Length > gameData.Length)
+                .SelectMany(dir =>
                 {
-                    var path = directory;
-                    while ((path != ksp.GameData()) & (path != null) & !newDirectories.Contains(path) )
+                    var results = new HashSet<string>();
+                    // remove the system paths, leaving everything after GameData/
+                    var relativeHead = KSPPathUtils.ToRelative(dir, gameData);
+                    var pathArray = relativeHead.Split('/');
+                    var builtPath = string.Empty;
+                    foreach (var path in pathArray)
                     {
-                        newDirectories.Add(path);
-                        DirectoryInfo parent = Directory.GetParent(path);
-                        path = parent.FullName;
+                        builtPath += path + '/';
+                        results.Add(KSPPathUtils.ToAbsolute(builtPath, gameData));
```

Comment:
Things can be installed to other directories. We should include any directory not under gameData without iterating for parents.